apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+/* global window */
```

Comment:
The goal of my previous comment about entry points vs js modules is to avoid making use of "window" at all inside library code, except to refer to browser-provided DOM apis. Attaching things to window as a method for sharing code makes it a lot harder to figure out module dependency graphs, since there is no explicit "import" required by client code.